gpytorch/module.py

```diff
@@ -114,6 +114,15 @@ def initialize(self, **kwargs):
 
         return self
 
+    def _load_raw_parameters(self, **kwargs):
+        for name, val in kwargs.items():
+            if '.' in name:
+                module, name = self._get_module_and_name(name)
+                module._load_raw_parameters(**{name: val})
+            else:
+                delattr(self, name)
+                setattr(self, name, val)
```

Comment:
this method is used to load values to model's parameters.

Reply:
`model.initialize(**hypers)` should accomplish this

Reply:
Unfortunately, `model.initialize(**hypers)` only copies data. Because I need to compute `grad` of potential function (loss) w.r.t. hypers, I need to overwrite model's attributes instead. Can I change how `initialize` method works to use overwriting instead of copying?

Reply:
Yeah, I think non breaking changes to the internals to make the interface cleaner are totally fine. By the same token, the transform Jacobians can probably live directly in the constraint objects, so that getting them all of them is simplified

Reply:
I think it will be a breaking change due to `expand_as`, `view_as` behaviour at [those lines](https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/module.py#L91), so I keep this as is.

Reply:
Do gpytorch.priors need log_det_abs_jacobian implementations or do those already inherit what they need from upstream? Maybe we could use Pyro distributions for non uniform priors?

Reply:
I don't think that gpytorch.priors need log_det_abs_jacobian. Pyro NUTS only needs to know how to transform from unconstrained space to constrained space; then it will adjust joint density based on that transform, regardless the prior which we put into the parameter. We'd need to make sure that the constraint for that parameter is a subset of the support of the prior though (to avoid drawing invalid values). So I think that something likes this will work well for `noise` parameter in NUTS:
```
likelihood = gpytorch.likelihoods.GaussianLikelihood(
    noise_constraint=gpytorch.constraints.GreaterThan(1e-3),
    noise_prior=gpytorch.priors.NormalPrior(0, 1)
)
```

Reply:
In Pyro, we use default constraint is the support of prior, so we don't need to specify `constraint` as in GPyTorch. I think that we can do it in GPyTorch too, which mean that if I specify
```
likelihood = gpytorch.likelihoods.GaussianLikelihood(
    noise_prior=gpytorch.priors.GammaPrior(0, 1)
)
```
then `noise_constraint` will be (by default) `gpytorch.constraints.Positive()`. But supporting that behaviour is not necessary though.